gobblin-compliance/src/main/java/gobblin/compliance/ComplianceRecord.java

```diff
@@ -0,0 +1,240 @@
+  public static final Splitter SLASH_SPLITTER = Splitter.on("/").omitEmptyStrings().trimResults();
+  public static final Splitter AT_SPLITTER = Splitter.on("@").omitEmptyStrings().trimResults();
+  public static final String SINGLE_QUOTE = "'";
+
+  private String complianceIdTable;
+  private String complianceIdentifier;
+  private String stagingDir;
+  private String stagingDb;
+  private Boolean commit;
+  private String jobId;
+  private DatasetDescriptor descriptor;
+
+  public void setPurgeQueries(List<String> purgeQueries) {
+    this.queries = purgeQueries;
+  }
+
+  public String getCompleteStagingTableName() {
+    StringBuilder sb = new StringBuilder();
+    sb.append(this.stagingDb);
+    sb.append(".");
+    sb.append(HivePurgerConfigurationKeys.STAGING_PREFIX);
+    sb.append(this.hivePartition.getTable().getTableName());
+    return sb.toString();
+  }
+
+  public String getCompleteOriginalTableName() {
+    StringBuilder sb = new StringBuilder();
+    sb.append(this.hivePartition.getTable().getDbName());
+    sb.append(".");
+    sb.append(this.hivePartition.getTable().getTableName());
+    return sb.toString();
+  }
+
+  public String getStagingTableLocation() {
+    return this.stagingDir + hivePartition.getTable().getTableName();
+  }
+
+  public String getStagingPartitionLocation() {
+    return getStagingTableLocation() + "/" + this.hivePartition.getName();
+  }
+
+  public String getFinalPartitionLocation() {
+    StringBuilder sb = new StringBuilder();
+    sb.append(this.hivePartition.getTable().getDataLocation());
+    sb.append("/" + this.jobId + "/");
+    sb.append(this.hivePartition.getName());
+    return sb.toString();
+  }
+
+  /**
+   * Add single quotes to the string, if not present.
+   * TestString will be converted to 'TestString'
+   * @param st
+   * @return
+   */
+  public String getQuotedString(String st) {
+    Preconditions.checkNotNull(st);
+    StringBuilder sb = new StringBuilder();
+    if (!st.startsWith(SINGLE_QUOTE)) {
+      sb.append(SINGLE_QUOTE);
+    }
+    sb.append(st);
+    if (!st.endsWith(SINGLE_QUOTE)) {
+      sb.append(SINGLE_QUOTE);
+    }
+    return sb.toString();
+  }
+
+  /**
+   * This method splits the multiple partitions separated by '/' to list.
+   * datepartition=2016-01-01-00/size=12345 will be [datepartition=2016-01-01-00, size=12345]
+   */
+  public List<String> getPartitionValues() {
+    return ImmutableList.<String>builder().addAll(SLASH_SPLITTER.splitToList(this.hivePartition.getName())).build();
+  }
+
+  /**
+   * This method splits the string on the basis of '=' into key value pairs and also add quotes to the values.
+   * [datepartition=2016-01-01-00, size=12345] will be [datepartition : '2016-01-01-00', size : '12345']
+   */
+  public Map<String, String> getPartitionMap() {
+    Map<String, String> partitionMap = new LinkedHashMap<>();
+    for (String st : getPartitionValues()) {
+      final List<String> list = EQUALITY_SPLITTER.splitToList(st);
```

Comment:
All "/" in the partition values are converted to %2 by Hive.
All "=" in the partition values are converted to %3 by Hive.